lang/java/reef-io/src/main/java/org/apache/reef/io/data/loading/api/DataLoadingRequestBuilder.java

```diff
@@ -217,12 +220,8 @@ public DataLoadingRequestBuilder setInputFormatClass(
    */
   @Deprecated
   public DataLoadingRequestBuilder setInputPath(final String inputPath) {
-    final DistributedDataSet dds = new DistributedDataSet();
-    dds.addPartition(DistributedDataSetPartition.newBuilder().setPath(inputPath)
-        .setLocation(DistributedDataSetPartition.LOAD_INTO_ANY_LOCATION)
-        .setDesiredSplits(Integer.valueOf(NumberOfDesiredSplits.DEFAULT_DESIRED_SPLITS)).build());
+    this.inputPath = inputPath;
     this.singleDataCenterStrategy = true;
```

Comment:
Users can call both `DataLoadingRequestBuilder.setInputPath()` and `DataLoadingRequestBuilder.setDistributedDataSet()` on the same builder, and the partition strategy class changes depending on which method was called first. However, if I understood correctly, the intended way of using this code is to call either `setInputPath()` or `setDistributedDataSet()`, not both. How about we throw an exception if both methods were called?